pilot/pkg/model/context.go

```diff
@@ -208,10 +210,10 @@ var DefaultXdsLogDetails = XdsLogDetails{}
 // Note: any errors returned will completely close the XDS stream. Use with caution; typically and empty
 // or no response is preferred.
 type XdsResourceGenerator interface {
-	Generate(proxy *Proxy, push *PushContext, w *WatchedResource, updates *PushRequest) (Resources, XdsLogDetails, error)
-
-	// GenerateDeltas returns the changed and removed resources, along with whether or not delta was actually used.
-	GenerateDeltas(proxy *Proxy, push *PushContext, updates *PushRequest, w *WatchedResource) (Resources, []string, XdsLogDetails, bool, error)
+	// Generate builds the xds resources for given proxy. Resources contain the newly added/updated resources.
+	// DeletedResources typically used in incremental Xds protocols contain the resources that have been deleted.
+	// It also indicates whether the response is delta or SoTW.
+	Generate(proxy *Proxy, push *PushContext, w *WatchedResource, updates *PushRequest) (Resources, DeletedResources, bool, XdsLogDetails, error)
```

Comment:
I almost wonder if we should return a struct as the response at this point... I am fine either way.

Reply:
I actually did this in the initial implementation but decided not to because it seemed too tedious. up to @ramaraochavali

Reply:
I agree it seems tedious to access from struct but I see @howardjohn's point. Let us leave it like this for now. We can do it later if we have add another return value.